File: Programing_Basics_with_JS/examPreparation/programmingBasicsOnlineRetakeExam2And3May2019/04.TouristShop.js
function touristShop(input) {
    let index = 0;
    let budget = Number(input[index]);
    index++;
    let command = input[index];
    index++;

    let productCount = 0;
    let totalCost = 0;

    while(command !== 'Stop') {
        let productName = command;
        let productPrice = Number(input[index]);
        index++;
        productCount++;

        if(productCount % 3 === 0) {
            productPrice = productPrice * 0.5;
            totalCost += productPrice;
        } else {
            totalCost += productPrice;
        }

        if(totalCost > budget) {
            break;
        }

        command = input[index];
        index++
    }

    let moneyDiff = Math.abs(budget - totalCost);

    if(budget >= totalCost) {
        console.log(`You bought ${productCount} products for ${totalCost.toFixed(2)} leva.`);
    } else {
        console.log("You don't have enough money!");
        console.log(`You need ${moneyDiff.toFixed(2)} leva!`);
    }
}

touristShop(["153.20",
"Backpack",
"25.20",
"Shoes",
"54",
"Sunglasses",
"30",
"Stop"])

touristShop(["54",
"Thermal underwear",
"24",
"Sunscreen",
"45"])

File: Programming_Basics_with_Python/programming_basics_online_retake_exam_2_and_3_may_2019/04._tourist_shop.py
budget = float(input())
command = input()

product_count = 0
all_product_price = 0
need_more_money = False
while not command == 'Stop':
    product_name = command
    product_price = float(input())
    product_count += 1

    if product_count % 3 == 0:
        product_price /= 2
    all_product_price += product_price

    if all_product_price > budget:
        product_count -= 1
        need_more_money = True
        break

    command = input()

if need_more_money:
    needed_money = all_product_price - budget
    print(f"You don't have enough money!")
    print(f'You need {needed_money:.2f} leva!')
else:
    print(f'You bought {product_count} products for {all_product_price:.2f} leva.')